apps/web/pages/team/[slug].tsx

```diff
@@ -249,29 +245,33 @@ function TeamPage({ team, isUnpublished, markdownStrippedBio, isValidOrgDomain }
 export const getServerSideProps = async (context: GetServerSidePropsContext) => {
   const ssr = await ssrInit(context);
   const slug = Array.isArray(context.query?.slug) ? context.query.slug.pop() : context.query.slug;
-  const { isValidOrgDomain } = orgDomainConfig(context.req.headers.host ?? "");
+  const { isValidOrgDomain, currentOrgDomain } = orgDomainConfig(
+    context.req.headers.host ?? "",
+    context.query.orgSlug
+  );
   const flags = await getFeatureFlagMap(prisma);
-
-  const team = await getTeamWithMembers(undefined, slug);
+  const team = await getTeamWithMembers({ slug, orgSlug: currentOrgDomain });
   const metadata = teamMetadataSchema.parse(team?.metadata ?? {});
-
   // Taking care of sub-teams and orgs
   if (
-    (isValidOrgDomain && team?.parent && !!metadata?.isOrganization) ||
     (!isValidOrgDomain && team?.parent) ||
     (!isValidOrgDomain && !!metadata?.isOrganization) ||
     flags["organizations"] !== true
   ) {
     return { notFound: true } as const;
   }
 
-  if (!team) {
+  if (!team || (team.parent && !team.parent.slug)) {
     const unpublishedTeam = await prisma.team.findFirst({
       where: {
-        metadata: {
-          path: ["requestedSlug"],
-          equals: slug,
-        },
+        ...(team?.parent
+          ? { id: team.parent.id }
```

Comment:
Can we have a possibility of unpublished team in an organization? It isn't handling that at the moment.

Reply:
No, subteams are always published as Org the tier includes them at no extra cost, what can be unpublished is just the Org for which you have to pay the Org tier